lib/src/data/model/response_model.dart

```diff
@@ -0,0 +1,21 @@
+import '../../../src/domain/entity/movie.dart';
+
+class ResponseModel {
+  late int page;
+  late int totalResults;
+  late int totalPages;
+  List<Movie> results = <Movie>[];
+
+  ResponseModel.fromJson(Map<String, dynamic> parsedJson) {
+    page = parsedJson['page'] ?? 0;
+    totalResults = parsedJson['total_results'] ?? 0;
+    totalPages = parsedJson['total_pages'] ?? 0;
+
+    if (parsedJson['results'] == null) return;
+
+    for (int i = 0; i < parsedJson['results']?.length; i++) {
+      final Movie result = Movie.fromJson(parsedJson['results'][i]);
+      results.add(result);
+    }
+  }
```

Comment:
Here instead to do the for iteration you can use something like this
`List<Movie>.from(json['results']).map((movie) => Movie.fromJson(movie))`

Reply:
done